### PR TITLE
chore(deps): Switch to yarn

### PR DESCRIPTION
**Pull request by lobsterkatie**
This switches to using `yarn` rather than `npm` for managing dependencies, since that's the standard practice at Sentry (see our repos for [the main sentry app](https://github.com/getsentry/sentry), our [JS SDK](https://github.com/getsentry/sentry-javascript), the [webpack plugin](https://github.com/getsentry/sentry-webpack-plugin), etc.).

Ref: WOR-2817

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,10 +19,10 @@ jobs:
         with:
           path: |
             **/node_modules
-          key: ${{ hashFiles('**/package-lock.json') }}
+          key: ${{ hashFiles('**/yarn.lock') }}
       - name: Install
         if: steps.cache-deps.outputs.cache-hit != 'true'
-        run: npm ci
+        run: yarn install
```
Comment (scttcper) [inline .github/workflows/build.yml:25]: this is closer to npm ci
```suggestion
        run: yarn install --frozen-lockfile
```

**Comment by scttcper** [comment]
we should probably get off node 10 and cache directly with the newer setup-node action instead of our own manual cache
https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

File: .github/workflows/build.yml (hunk 1 above)
Comment (lobsterkatie) [reply .github/workflows/build.yml:25]: It's true. Given that our only non-dev dependency is `@sentry/cli`, though, I wasn't sure it was worth the complication. I can add it if you think it's important.

**Comment by lobsterkatie** [comment]
> we should probably get off node 10 and cache directly with the newer setup-node action instead of our own manual cache [actions/setup-node@`main`/docs/advanced-usage.md#caching-packages-data](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md?rgh-link-date=2023-03-06T17%3A44%3A07Z#caching-packages-data)

Agreed. One change at a time, though. 🙂

**Review by scttcper — APPROVED** [review]

**Pull request merged**